packages/integration-sdk-runtime/src/execution/config.ts

```diff
@@ -70,6 +73,10 @@ function convertEnvironmentVariableValueForField(
       }
       break;
     }
+    case 'json': {
+      convertedValue = JSON.parse(environmentVariableValue);
```

Comment:
If we fail to parse the value, it will probably be a good idea to throw an error saying that value `foo` is not valid JSON rather than just bubbling up a generic JSON.parse error.